### PR TITLE
Add profiling for DiceDB

### PR DESCRIPTION
**Pull request by soumya-codes**
This PR adds profiling support for DiceDB

How to run profiling:

```
go build main.go  
./main --enable-multithreading=true --enable-profiling=true
```

### CODE REVIEW

**Comment by soumya-codes** [comment]
@arpitbbhayani, PR for profiling DiceDB

File: main.go [hunk 1]
```diff
@@ -153,6 +157,61 @@ func main() {
 			}
 		}()
 	} else {
+		if config.EnableProfiling {
```
Comment (lucifercr07) [inline main.go:160]: Shall we move this to another func like `func setupProfiling(logr *slog.Logger)`?

File: main.go [hunk 2]
```diff
@@ -153,6 +157,61 @@ func main() {
...
+			// Start CPU profiling
+			cpuFile, err := os.Create("cpu.prof")
+			if err != nil {
+				logr.Warn("could not create file cpu.prof: ", slog.Any("error", err))
+			}
+			defer cpuFile.Close()
```
Comment (lucifercr07) [inline main.go:166]: Would this lead to panic if `Create` failed?

File: main.go [hunk 3]
```diff
@@ -153,6 +157,61 @@ func main() {
...
+					logr.Warn("could not write block profile: ", slog.Any("error", err))
+				}
+			}()
+
+			// Start execution trace
+			traceFile, err := os.Create("trace.out")
+			if err != nil {
+				logr.Warn("could not create trace output file: ", slog.Any("error", err))
+			}
+			defer traceFile.Close()
```
Comment (lucifercr07) [inline main.go:207]: Same as above

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (soumya-codes) [reply main.go:160]: Yah, let me organize the code better, lazily just pushed what I had written for profiling.

**Comment by soumya-codes** [comment]
@lucifercr07 have implemented your suggestions. Please let me know how they look.

**Review by lucifercr07 — APPROVED** [review]
LGTM

File: main.go [hunk 4]
```diff
@@ -36,6 +39,7 @@ func init() {
 	flag.IntVar(&config.KeysLimit, "keys-limit", config.KeysLimit, "keys limit for the dicedb server. "+
 		"This flag controls the number of keys each shard holds at startup. You can multiply this number with the "+
 		"total number of shard threads to estimate how much memory will be required at system start up.")
+	flag.BoolVar(&config.EnableProfiling, "enable-profiling", false, "run server in multithreading mode")
```
Comment (lucifercr07) [inline main.go:42]: Shall we update this message?

Comment (soumya-codes) [reply main.go]: updated. thanks for catching.

**Comment by vinitparekh17** [comment]
@soumya-codes What is the purpose of pushing the binary file "main"?

**Comment by lucifercr07** [comment]
> @soumya-codes What is the purpose of pushing the binary file "main"?

That's a miss, will raise a PR to remove it. Thanks for pointing it out.

**Comment by soumya-codes** [comment]
@vinitparekh17 thanks for catching. I see @lucifercr07 has already raised a PR for this.

**Pull request merged**